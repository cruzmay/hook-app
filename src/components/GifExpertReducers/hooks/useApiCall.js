import { useEffect, useState } from "react"
import { ApiCall } from "../helpers/ApiCall"

export const useApiCall = (category) => {
   
    const [state, setstate] = useState({
        data: [],
        loading: true
    })

    useEffect(()=>{
        ApiCall(category).then( img => 
            setstate({
                data: img,
                loading: false
            })
        )
    },[category])

    return state

}
