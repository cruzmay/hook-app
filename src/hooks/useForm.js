import { useState } from "react"

export const useForm = (initialState={}) => {
    
    const [formState, setformState] = useState(initialState)


    const handleInputChange = ({target}) => {
        
        setformState({
            ...formState,
            [target.name]: target.value
        })
    }
    return [formState, handleInputChange]
}
