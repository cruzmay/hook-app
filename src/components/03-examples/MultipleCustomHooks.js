import React from 'react';
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch';
import './multiple.css'

export const MultipleCustomHooks = () => {

    const {counter, increment} = useCounter(1)

    const {loading, data} = useFetch(`https://breakingbadapi.com/api/quotes/${counter}`)
    
    const {author, quote} = !!data && data[0]
    
    console.log(author, quote)
    return (
        <div>
            <h1>Breaking Bad quotes</h1>
            <hr/>
            {
                loading ?
                (
                    <div className="alert alert-info text-center">
                        loading ...
                    </div>
                ):
                (
                    <div className="blockquote text-right">
                        <p className="mb-3">{quote}</p>
                        <footer className="blockquote-footer">{author}</footer>
                    </div>
                )

                
            }
            <button 
            className="btn btn-primary"
            onClick = {increment}
            >next quote
            </button>
        </div>
    )
}
