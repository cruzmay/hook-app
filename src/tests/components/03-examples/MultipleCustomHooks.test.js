import { shallow } from 'enzyme';
import React from 'react';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('haciendo pruebas en el componente <MultipleCustomHooks/>', () => {
    
    
    
    test('debe mostrarse correctamente ', () => {    

        useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    })

        useFetch.mockReturnValue({
            data:  null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks/>)
        expect(wrapper).toMatchSnapshot()
    })
    test('debe de mostrar la informacion', () => {

        useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    })
        
        useFetch.mockReturnValue({
            data:  [{
                author: 'fernando',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks/>)
        console.log(wrapper.html())
        expect(wrapper.find('.alert').exists()).toBe(false)
        expect(wrapper.find('.mb-3').text().trim()).toBe('Hola Mundo')
        expect(wrapper.find('footer').text().trim()).toBe('fernando')
        expect(wrapper).toMatchSnapshot()

    })
    
})
