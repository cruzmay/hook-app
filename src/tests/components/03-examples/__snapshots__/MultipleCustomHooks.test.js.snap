// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`haciendo pruebas en el componente <MultipleCustomHooks/> debe de mostrar la informacion 1`] = `
<div>
  <h1>
    Breaking Bad quotes
  </h1>
  <hr />
  <div
    className="blockquote text-right"
  >
    <p
      className="mb-3"
    >
      Hola Mundo
    </p>
    <footer
      className="blockquote-footer"
    >
      fernando
    </footer>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    next quote
  </button>
</div>
`;

exports[`haciendo pruebas en el componente <MultipleCustomHooks/> debe mostrarse correctamente  1`] = `
<div>
  <h1>
    Breaking Bad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    loading ...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    next quote
  </button>
</div>
`;
